airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationDefinitionsHandler.java

```diff
@@ -124,6 +126,14 @@ public DestinationDefinitionReadList listLatestDestinationDefinitions() {
     }
   }
 
+  public DestinationDefinitionReadList listDestinationDefinitionsForWorkspace(final WorkspaceIdRequestBody workspaceIdRequestBody)
+      throws IOException {
+    return toDestinationDefinitionReadList(MoreLists.concat(
+        configRepository.listPublicDestinationDefinitions(false),
+        configRepository.listGrantedDestinationDefinitions(
```

Comment:
I was trying to think through what would happen if a definition was somehow in both of these lists, like in a situation where a `public: false` definition later became `public: true`. But then I realized you're already  filtering out `public: true` definitions in the granted query, so I think we're covered there!

Reply:
oh right exactly!  I asked about whether we should clean up old grants when I first thought of this too, but for now I think filtering out public in the granted query should be fine.